daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -2269,7 +2333,7 @@ func NewDeviceSet(root string, doInit bool, options []string, uidMaps, gidMaps [
 		metaDataLoopbackSize:  defaultMetaDataLoopbackSize,
 		baseFsSize:            defaultBaseFsSize,
 		overrideUdevSyncCheck: defaultUdevSyncOverride,
-		filesystem:            "ext4",
+		defaultFilesystem:     determineDefaultFS(),
```

Comment:
we shouldn't need to call this at creation, but only if user did not specify `dm.fs`, so after the options are parsed and checked if `devices.userFilesystem` is not set